File: examples/websocket/src/toppage_handler.erl
%% Feel free to use, reuse and abuse the code in this file.

-module(toppage_handler).

-export([init/3]).
-export([handle/2]).
-export([terminate/3]).

init(_Transport, Req, []) ->
	{ok, Req, undefined}.

handle(Req, State) ->
	Html = get_html(),
	{ok, Req2} = cowboy_req:reply(200,
		[{<<"content-type">>, <<"text/html">>}],
		Html, Req),
	{ok, Req2, State}.

terminate(_Reason, _Req, _State) ->
	ok.

get_html() ->
	{ok, Cwd} = file:get_cwd(),
	Filename =filename:join([Cwd, "priv", "html_ws_client.html"]),
	{ok, Binary} = file:read_file(Filename),
	Binary.

File: examples/elixir_websocket/lib/elixir_websocket/top_page_handler.ex
# Feel free to use, reuse and abuse the code in this file.

defmodule ElixirWebsocket.TopPageHandler do
  def init(_transport, req, []) do
    {:ok, req, nil}
  end

  def get_html do
    {:ok, cwd} = :file.get_cwd()
    filename = :filename.join([cwd, "priv", "html_ws_client.html"])
    {:ok, binary} = :file.read_file(filename)
    binary
  end

  def handle(req, state) do
  html = get_html()
    {:ok, req} = :cowboy_req.reply(200,
    [{<<"content-type">>, <<"text/html">>}],
      html, req)
    {:ok, req, state}
  end

  def terminate(_reason, _req, _state) do
    :ok
  end
end